File: dungeons-1.21.11/src/main/java/fishmod/utils/config/values/Floor7.java
package fishmod.utils.config.values;

import config.practical.data.SoundData;
import config.practical.manager.ConfigValue;
import net.minecraft.sound.SoundEvents;

public class Floor7 {

    @ConfigValue
    public static boolean enableBossWaypoints = false;

    @ConfigValue
    public static int nextWaypointColor = 0xff00F7F7;

    @ConfigValue
    public static boolean nextWaypointThroughWall = false;

    @ConfigValue
    public static boolean enableCrystalSpawnTime = false;

    @ConfigValue
    public static boolean enableStormTickTimer = false;

    @ConfigValue
    public static int stormTickTimerColor = 0xffffffff;

    @ConfigValue
    public static boolean tickDownStormTickTimer = false;

    @ConfigValue
    public static boolean enableStormDeathTime = false;

    // LB (Last Breath) release window — countdown shown from 30s until 34.35s on the Storm (P2)
    // clock, telling you when to shoot Last Breath.
    @ConfigValue
    public static boolean enableLbReleaseTimer = false;

    @ConfigValue
    public static int lbReleaseTimerColor = 0xffff5555;

    @ConfigValue
    public static boolean notifyUsedSpiritMask = false;

    @ConfigValue
    public static boolean displayDistanceToLedge = false;

    @ConfigValue
    public static boolean enableGoldorTickTimer = false;

    @ConfigValue
    public static boolean inDeathTicks = true;

    @ConfigValue
    public static boolean enableTermStartTimer = false;

    @ConfigValue
    public static boolean enablePositionalMessages = false;

    @ConfigValue
    public static boolean enableRelicStartTimer = false;

    @ConfigValue
    public static int relicSpawnTicks = 42;

    @ConfigValue
    public static boolean enableRelicPlaceTime = false;

    @ConfigValue
    public static boolean renderRelicHighlight = false;

    @ConfigValue
    public static boolean blockIncorrectRelicPlace = false;

    @ConfigValue
    public static boolean replaceWithProgressBar = false;

    @ConfigValue
    public static boolean useValleyBar = true;

    @ConfigValue
    public static boolean combineTickTimers = false;

    @ConfigValue
    public static boolean dragSpawnTimers = false;

    @ConfigValue
    public static boolean sendSoundOnDragSpawn = false;

    @ConfigValue
    public static boolean displayLocationNotification = false;

    @ConfigValue
    public static int notificationDuration = 15;

    @ConfigValue
    public static SoundData atLocationSound = new SoundData(SoundEvents.BLOCK_NOTE_BLOCK_PLING.value(), 1, 1);

    @ConfigValue
    public static int notificationRepetitions = 3;

    @ConfigValue
    public static boolean predevForAll = true;

    @ConfigValue
    public static boolean showAllRelicTimes = true;

    @ConfigValue
    public static boolean notifyPre4Completion = false;

    @ConfigValue
    public static boolean notifyStormCrush = false;

    @ConfigValue
    public static boolean timePillarExplosion = false;

    @ConfigValue
    public static boolean notifiyMelody = false;

    // Counts down 3.75s from Goldor's health hitting 0, then shows a "LEAP!" title + sound.
    @ConfigValue
    public static boolean leapNotifications = false;

    @ConfigValue
    public static boolean disableTitlesAtPre4 = false;

    @ConfigValue
    public static boolean showDistanceAtYellowOnly = false;

    @ConfigValue
    public static boolean dontNotifiyForYourself = true;

    @ConfigValue
    public static boolean hideTerminalTitles = false;

    @ConfigValue
    public static boolean terminalTimeStamps = false;

    @ConfigValue
    public static boolean crystalPlaceReminder = false;

    @ConfigValue
    public static boolean showSectionProgress = false;

    @ConfigValue
    public static boolean notifySSCompletion = false;

    @ConfigValue
    public static boolean disableTitlesAtSS = false;

    @ConfigValue
    public static boolean instantlyDisplayCrystalReminder = false;

    @ConfigValue
    public static boolean dragonHealth = false;

    @ConfigValue
    public static boolean maxorStunDuration = false;

    @ConfigValue
    public static boolean makeGoldorTickUp = true;

    @ConfigValue
    public static boolean capitalizeHealthNumbers = true;

    @ConfigValue
    public static boolean sectionCompletionNotification = false;

    @ConfigValue
    public static SoundData sectionChangeSound = new SoundData(SoundEvents.BLOCK_NOTE_BLOCK_PLING.value(), 0, 1);

    @ConfigValue
    public static boolean enableMaxorTickTimer = false;

    @ConfigValue
    public static boolean showCurrentSection = false;

    @ConfigValue
    public static boolean sectionColorProgress = false;

    @ConfigValue
    public static boolean sectionPrevObjective = false;

    @ConfigValue
    public static boolean dragonTracer = false;

    @ConfigValue
    public static boolean assumeCore = true;

    @ConfigValue
    public static boolean assumeSplitEE2 = false;
}

File: dungeons-26.1.2/src/main/kotlin/fishmod/utils/config/values/Floor7.kt
package fishmod.utils.config.values

import fishmod.shaded.practicalconfig.data.SoundData
import fishmod.shaded.practicalconfig.manager.ConfigValue
import net.minecraft.sounds.SoundEvents

object Floor7 {

    @ConfigValue
    @JvmField
    var enableCrystalSpawnTime: Boolean = false

    @ConfigValue
    @JvmField
    var enableStormTickTimer: Boolean = false

    @ConfigValue
    @JvmField
    var stormTickTimerColor: Int = 0xffffffff.toInt()

    @ConfigValue
    @JvmField
    var tickDownStormTickTimer: Boolean = false

    @ConfigValue
    @JvmField
    var enableStormDeathTime: Boolean = false

    @ConfigValue
    @JvmField
    var enableLbReleaseTimer: Boolean = false

    @ConfigValue
    @JvmField
    var lbReleaseTimerColor: Int = 0xffff5555.toInt()

    @ConfigValue
    @JvmField
    var lbReleaseTimerPingMs: Int = 0

    @ConfigValue
    @JvmField
    var enablePyTimer: Boolean = false

    @ConfigValue
    @JvmField
    var pyTimerColor: Int = 0xffff55ff.toInt()

    @ConfigValue
    @JvmField
    var pyTimerPingMs: Int = 0

    @ConfigValue
    @JvmField
    var enableNecronLbTimer: Boolean = false

    @ConfigValue
    @JvmField
    var necronLbPingMs: Int = 0

    @ConfigValue
    @JvmField
    var necronLbColor: Int = 0xffff5555.toInt()

    @ConfigValue
    @JvmField
    var enableGoldorTickTimer: Boolean = false

    @ConfigValue
    @JvmField
    var inDeathTicks: Boolean = true

    @ConfigValue
    @JvmField
    var enableTermStartTimer: Boolean = false

    @ConfigValue
    @JvmField
    var enableRelicStartTimer: Boolean = false

    @ConfigValue
    @JvmField
    var relicSpawnTicks: Int = 42

    @ConfigValue
    @JvmField
    var renderRelicHighlight: Boolean = false

    @ConfigValue
    @JvmField
    var relicTimesEnabled: Boolean = false

    @ConfigValue
    @JvmField
    var relicTimesParty: Boolean = false

    @ConfigValue
    @JvmField
    var notifyPre4Completion: Boolean = false

    @ConfigValue
    @JvmField
    var notifyStormCrush: Boolean = false

    @ConfigValue
    @JvmField
    var timePillarExplosion: Boolean = false

    @ConfigValue
    @JvmField
    var notifiyMelody: Boolean = false

    @ConfigValue
    @JvmField
    var disableTitlesAtPre4: Boolean = false

    @ConfigValue
    @JvmField
    var hideTerminalTitles: Boolean = false

    @ConfigValue
    @JvmField
    var terminalTimeStamps: Boolean = false

    @ConfigValue
    @JvmField
    var crystalPlaceReminder: Boolean = false

    @ConfigValue
    @JvmField
    var showSectionProgress: Boolean = false

    @ConfigValue
    @JvmField
    var notifySSCompletion: Boolean = false

    @ConfigValue
    @JvmField
    var disableTitlesAtSS: Boolean = false

    @ConfigValue
    @JvmField
    var instantlyDisplayCrystalReminder: Boolean = false

    @ConfigValue
    @JvmField
    var makeGoldorTickUp: Boolean = true

    @ConfigValue
    @JvmField
    var capitalizeHealthNumbers: Boolean = true

    @ConfigValue
    @JvmField
    var sectionCompletionNotification: Boolean = false

    @ConfigValue
    @JvmField
    var sectionChangeSound: SoundData = SoundData(SoundEvents.NOTE_BLOCK_PLING.value(), 0f, 1f)

    @ConfigValue
    @JvmField
    var enableTickTimers: Boolean = true

    @ConfigValue
    @JvmField
    var enableMaxorTickTimer: Boolean = false

    @ConfigValue
    @JvmField
    var showCurrentSection: Boolean = false

    @ConfigValue
    @JvmField
    var sectionColorProgress: Boolean = false

    @ConfigValue
    @JvmField
    var sectionPrevObjective: Boolean = false

    @ConfigValue
    @JvmField
    var playersLeapedEnabled: Boolean = false

    @ConfigValue
    @JvmField
    var playersLeapedAnyClass: Boolean = false

    @ConfigValue
    @JvmField
    var gateDisplayEnabled: Boolean = false

    @ConfigValue
    @JvmField
    var gateDisplayScale: Float = 6f

    @ConfigValue
    @JvmField
    var bloodSolverEnabled: Boolean = false

    @ConfigValue
    @JvmField
    var bloodSolverKillTitle: Boolean = false

    @ConfigValue
    @JvmField
    var bloodSolverSpeedAlert: Boolean = false

    @ConfigValue
    @JvmField
    var bloodSolverSpeedAlertParty: Boolean = false

    @ConfigValue
    @JvmField
    var bloodSolverDecimals: Int = 1

    @ConfigValue
    @JvmField
    var bloodSolverBoxColor: Int = 0xffff00ff.toInt()

    @ConfigValue
    @JvmField
    var bloodSolverLineColor: Int = 0xff55ffff.toInt()

    @ConfigValue
    @JvmField
    var bloodSolverTimerColor: Int = 0xffffffff.toInt()
}
